### ISSUE TITLE
invisible mouse cursor in gamescope

### ISSUE THREAD

**Issue by unicornsandcandies** [ublue-os/bazzite#1354]
### Describe the bug

To replicate: 
Hold steam button and use right touchpad to simulate mouse 

Cursor is invisible in every game that doesn't theme it by their own

### What did you expect to happen?

I expected mouse cursor to show up, so I can choose options in a game launcher (without using the touchscreen)

### Output of `rpm-ostree status`

```shell
State: idle
Deployments:
● ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-deck-gnome:stable
                   Digest: sha256:c53a8b7cdb78233b3b13f58a6f770b18a0abb8f6f283e9cf4cf33d3c8f552a74
                  Version: 40.20240712.0 (2024-07-12T04:58:14Z)
          LayeredPackages: plymouth-theme-script
            LocalPackages: pedro-raccoon-1.1-1.1.x86_64

  ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-deck-gnome:stable
                   Digest: sha256:4194afb903fc79b1028d4f3140f694c2a220497678405a72e3de3e3ff3b62b27
                  Version: 40.20240710.0 (2024-07-11T02:46:00Z)
          LayeredPackages: plymouth-theme-script
            LocalPackages: pedro-raccoon-1.1-1.1.x86_64
```

### Hardware

Galileo
Steam Deck OLED

**Comment by WuffyTopside** [comment]
Same issue in Bazzite Deck KDE on Deck LCD. Might be a problem with the underlying Fedora image. That, or it might be Steam not handling the mouse info properly.

**Comment by unicornsandcandies** [comment]
> Same issue in Bazzite Deck KDE on Deck LCD. Might be a problem with the underlying Fedora image. That, or it might be Steam not handling the mouse info properly.

I don't know if it happened for you but the cursor was weird already for some time. It was black and overly big. Not like the theme I've set in the desktop mode.

**Comment by WuffyTopside** [comment]
My cursor is fine in desktop mode. People are saying it might be a Wayland issue, so I'm going to test it.

**Comment by unicornsandcandies** [comment]
> My cursor is fine in desktop mode. People are saying it might be a Wayland issue, so I'm going to test it.

Isn't the desktop mode also using Wayland on deck? I mean, I'm on gnome deck and the cursor is fine

**Comment by WuffyTopside** [comment]
Yes, they both use Wayland. From what I saw, the issue is how Steam talks to Wayland, or something along those lines at least.

**Comment by unicornsandcandies** [comment]
Update: 
It works when docked for some weird reason 

![20240713_205047](https://github.com/user-attachments/assets/654d69ba-609e-4785-9810-584d1fc09946)
![20240713_205031](https://github.com/user-attachments/assets/4cf9091d-a44c-4523-879e-c325213ba7e0)

**Comment by WuffyTopside** [comment]
Odd. I'm not sure what's going on.

**Comment by KyleGospo** [comment]
I'm not able to reproduce this problem on my OLED, but I'm running the GNOME image. Let me dig a little bit and I might have questions for you.

**Comment by KyleGospo** [comment]
Can one of you give me the contents of:
`$XDG_CONFIG_HOME/gtk-3.0/settings.ini`

**Comment by WuffyTopside** [comment]
Unfortunately, I'm unable to provide file contents of any kind. I ended up just installing Chimera OS because I lack patience and forethought. I apologize.

**Comment by unicornsandcandies** [comment]
> Can one of you give me the contents of: `$XDG_CONFIG_HOME/gtk-3.0/settings.ini`

deck@fedora:~$ $XDG_CONFIG_HOME/gtk-3.0/settings.ini
-bash: /gtk-3.0/settings.ini: Nie ma takiego pliku ani katalogu
deck@fedora:~$ cat .config/gtk-3.0/settings.ini
[Settings]
gtk-application-prefer-dark-theme=0
deck@fedora:~$

**Comment by WuffyTopside** [comment]
Here's the contents of ~/.config/gtk-3.0/settings.ini from a fresh install. The only change made is to the theme, since my taskbar decided it wanted to play hide-and-seek with me.

`[Settings]
gtk-application-prefer-dark-theme=true
gtk-button-images=true
gtk-cursor-theme-name=breeze_cursors
gtk-cursor-theme-size=24
gtk-decoration-layout=icon:minimize,maximize,close
gtk-enable-animations=true
gtk-font-name=Noto Sans,  11
gtk-icon-theme-name=breeze-dark
gtk-menu-images=true
gtk-modules=colorreload-gtk-module
gtk-primary-button-warps-slider=true
gtk-sound-theme-name=ocean
gtk-toolbar-style=3
gtk-xft-dpi=172032`

**Comment by jpwchang** [comment]
@KyleGospo For the record I'm also on the GNOME image and also seeing this exact problem.

**Comment by WuffyTopside** [comment]
There's a way to get a cursor until this is fixed. Just switch to desktop mode and launch the games that use a cursor from there.

**Comment by jpwchang** [comment]
I checked `gamescope-stdout.log` and found the same `libei`-related error that was reported over in discord as a possible clue:

`[gamescope] [[0;31mError[0m] [0;37mwlserver:[0m Gamescope built without libei, XTEST will not be available!`

**Comment by KyleGospo** [comment]
> I checked `gamescope-stdout.log` and found the same `libei`-related error that was reported over in discord as a possible clue:
> 
> `[gamescope] [[0;31mError[0m] [0;37mwlserver:[0m Gamescope built without libei, XTEST will not be available!`

The newest build has this restored. Let me know if updating fixes it for you.

**Comment by jpwchang** [comment]
@KyleGospo I pasted that log message right after updating to `bazzite-deck-gnome 40.20240714.0` this morning. Or is there a newer build than that?

**Comment by KyleGospo** [comment]
> @KyleGospo I pasted that log message right after updating to `bazzite-deck-gnome 40.20240714.0` this morning. Or is there a newer build than that?

Turning it back on broke gnome for me, I'm interested in KDE testing.

**Comment by jpwchang** [comment]
Makes sense, I'm perfectly happy to wait a while longer since this doesn't really affect my workflow all that much. The more important point was that I just wanted to flag the fact that this evidently isn't a KDE-only issue!

**Comment by WuffyTopside** [comment]
I can check KDE when the fix drops. When it does, @ me so I know to check it.

**Comment by jpwchang** [comment]
Happy to report that after running another update the issue appears resolved, at least here on GNOME! Perhaps @WuffyTopside can confirm on KDE?

Thanks for the quick fix!

**Comment by WuffyTopside** [comment]
I'm happy to report that the cursor is now visible in Gamescope on KDE! Thanks @jpwchang , and to the whole Bazzite dev team!

**Comment by xsxavier** [comment]
This issue is still happening currently on gamescope. Often games have an invisible cursor, sometimes using a mouse instead of the controller will fix it.
